### ISSUE TITLE
Allow field "reload_cmd" to contain the @-symbol

### ISSUE THREAD

**Issue by mgeha** [Froxlor/Froxlor#1175]
I am using gentoo with different php-fpm versions and configs. I use systemd service templates to configure those php-fpm services.
The command to restart would be
`/bin/systemctl restart php-fpm@8.1 `

Today I tired to edit php-fpm versions in froxlor 2.0.22 - but I cannot save them I the command contains the @-symbol. Please allow for this.

Thx in advance

**Comment by d00p** [comment]
As workaround, you can specify a custom shell script that runs the desired command

**Comment by mgeha** [comment]
Well, I just edited reload_cmd in the froxlor mysql database, but I think we can agree that both variants are not very elegant :-)

**Comment by d00p** [comment]
Try the followin patch, it should resolve the issue:

```diff
diff --git a/lib/Froxlor/Api/Commands/FpmDaemons.php b/lib/Froxlor/Api/Commands/FpmDaemons.php
index 26d4d493..d74daa6d 100644
--- a/lib/Froxlor/Api/Commands/FpmDaemons.php
+++ b/lib/Froxlor/Api/Commands/FpmDaemons.php
@@ -202,7 +202,7 @@ class FpmDaemons extends ApiCommand implements ResourceEntity

                        // validation
                        $description = Validate::validate($description, 'description', Validate::REGEX_DESC_TEXT, '', [], true);
-                       $reload_cmd = Validate::validate($reload_cmd, 'reload_cmd', '/^[a-z0-9\/\._\- ]+$/i', '', [], true);
+                       $reload_cmd = Validate::validate($reload_cmd, 'reload_cmd', '/^[a-z0-9\/\._\-@ ]+$/i', '', [], true);
                        $sel_stmt = Database::prepare("SELECT `id` FROM `".TABLE_PANEL_FPMDAEMONS."` WHERE `reload_cmd` = :rc");
                        $dupcheck = Database::pexecute_first($sel_stmt, ['rc' => $reload_cmd]);
                        if ($dupcheck && $dupcheck['id']) {
@@ -327,7 +327,7 @@ class FpmDaemons extends ApiCommand implements ResourceEntity

                        // validation
                        $description = Validate::validate($description, 'description', Validate::REGEX_DESC_TEXT, '', [], true);
-                       $reload_cmd = Validate::validate($reload_cmd, 'reload_cmd', '/^[a-z0-9\/\._\- ]+$/i', '', [], true);
+                       $reload_cmd = Validate::validate($reload_cmd, 'reload_cmd', '/^[a-z0-9\/\._\-@ ]+$/i', '', [], true);
                        $sel_stmt = Database::prepare("SELECT `id` FROM `".TABLE_PANEL_FPMDAEMONS."` WHERE `reload_cmd` = :rc");
                        $dupcheck = Database::pexecute_first($sel_stmt, ['rc' => $reload_cmd]);
                        if ($dupcheck && $dupcheck['id'] != $id) {
```

**Comment by d00p** [comment]
fixed in 9d2077ddee366ea301490d4d2d82892b22632aed
